### DISCUSSION TITLE
Rauthy License

### DISCUSSION THREAD

**Discussion by sebadob** [sebadob/rauthy#127] [Announcements]
A quick info about the current choice of the AGPLv3.

I am aware of the fact, that the AGPL limits the possibilities to integrate it with other open source projects.  
Rauthy may be changed to an Apache 2.0 or similar in the future, I am not fully sold on the AGPLv3 yet.  
The AGPLv3 was chosen in the beginning to give myself some more freedom as a single developer, since I did not have any idea, how fast I could get it up an running to a point, where I would be happy with the feature set.

**Comment by damooo** [comment]
Does using `Rauthy` as a git dependency in cargo projects with MIT/APACHE license, violates the license? 

> Rauthy may be changed to an Apache 2.0 or similar in the future

That would be really great for seamless and flexible integrations. Perticularly as rust code being compiled.

**Reply by erlend-sh** [reply]
> Does using Rauthy as a git dependency in cargo projects with MIT/APACHE license, violates the license?

[Yes](https://opensource.stackexchange.com/questions/1640/if-im-using-a-gpl-3-library-in-my-project-can-i-license-my-project-under-mit-l).

**Reply by sebadob** [reply]
I actually decided to switch Rauthy to an Apache 2.0 with the next release.  
This will just make a lot of things easier.

**Reply by MinisculeGirraffe** [reply]
Nice!

Apache or MIT license would be a great change.
